### PR TITLE
Fix issue with ServiceNow incidents not including link to access request

### PR DESCRIPTION
**Pull request by EdwardDowling**
Fixes link to access request in incident description, plugin status on plugin page, and incidents now get the caller field filled if possible.

Status sink added to client is set in /e PR [here](https://github.com/gravitational/teleport.e/pull/2468)

### CODE REVIEW

File: integrations/access/servicenow/client.go [hunk 1]
```diff
@@ -250,6 +260,24 @@ func (snc *Client) CheckHealth(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 	defer resp.RawResponse.Body.Close()
+
+	defer func() {
```
Comment (marcoandredinis) [inline integrations/access/servicenow/client.go:264]: Why do we need to defer the `Emit` operation?

Comment (marcoandredinis) [inline integrations/access/servicenow/client.go:264]: Why do we need to defer the `Emit` operation?

File: integrations/access/servicenow/client.go [hunk 2]
```diff
@@ -250,6 +260,24 @@ func (snc *Client) CheckHealth(ctx context.Context) error {
...
+			case resp.StatusCode() == http.StatusUnauthorized:
+				code = types.PluginStatusCode_UNAUTHORIZED
+			case resp.StatusCode() >= 200 && resp.StatusCode() < 400:
+				code = types.PluginStatusCode_RUNNING
+			default:
+				code = types.PluginStatusCode_OTHER_ERROR
+			}
+			if err := snc.StatusSink.Emit(ctx, &types.PluginStatusV1{Code: code}); err != nil {
+				log := logger.Get(resp.Request.Context())
+				log.WithError(err).Errorf("Error while emitting servicenow plugin status: %v", err)
```
Comment (marcoandredinis) [inline integrations/access/servicenow/client.go:277]: Should we also log the `code`? Could be useful when looking at the logs

**Review by marcoandredinis — APPROVED** [review]

File: integrations/access/servicenow/client.go (hunk 1 above)
Comment (EdwardDowling) [reply integrations/access/servicenow/client.go]: We dont, have removed the defer. Can't remember why i did that.

File: integrations/access/servicenow/client.go [hunk 3]
```diff
@@ -250,6 +260,24 @@ func (snc *Client) CheckHealth(ctx context.Context) error {
...
+		case resp.StatusCode() >= 200 && resp.StatusCode() < 400:
+			code = types.PluginStatusCode_RUNNING
+		default:
+			code = types.PluginStatusCode_OTHER_ERROR
+		}
+		log := logger.Get(resp.Request.Context())
+		log.Infof("Emitting servicenow plugin status: %v", code)
+		if err := snc.StatusSink.Emit(ctx, &types.PluginStatusV1{Code: code}); err != nil {
+			log.WithError(err).Errorf("Error while emitting servicenow plugin status: %v", err)
+		}
```
Comment (marcoandredinis) [inline integrations/access/servicenow/client.go:278]: Sorry, I was not clear
My suggestion is to log the code only when it failed to `Emit`

Always logging seems a bit noisy (could be ok for debug level, tho).

So, something like this:
```suggestion
		if err := snc.StatusSink.Emit(ctx, &types.PluginStatusV1{Code: code}); err != nil {
			log := logger.Get(resp.Request.Context())
			log.WithError(err).WithField("code", resp.StatusCode()).Errorf("Error while emitting servicenow plugin status: %v", err)
		}
```

File: integrations/access/servicenow/client_test.go [hunk 4]
```diff
@@ -57,8 +57,9 @@ func TestCreateIncident(t *testing.T) {
 	assert.NoError(t, err)
 
 	expected := Incident{
-		ShortDescription: "Access request from someUser",
-		Description:      "someUser requested permissions for roles role1, role2 on Teleport at 01 Jan 01 00:00 UTC.\nReason: someReason\n\n",
+		ShortDescription: "Teleport access request from user someUser",
+		Description:      "Teleport user someUser submitted access request for roles (or resources, for resource-based requests) role1, role2 on Teleport cluster .\nReason: someReason\n\n",
```
Comment (r0mant) [inline integrations/access/servicenow/client_test.go:61]:
```suggestion
		Description:      "Teleport user someUser submitted access request for roles role1, role2 on Teleport cluster .\nReason: someReason\n\n",
```

**Review by r0mant — APPROVED** [review]
I think you need to fix the test as well but otherwise lgtm

**Pull request merged**